moto/swf/models/history_event.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import unicode_literals
+from datetime import datetime
+from time import mktime
+
+from ..utils import decapitalize, now_timestamp
+
+
+class HistoryEvent(object):
+    def __init__(self, event_id, event_type, **kwargs):
+        self.event_id = event_id
+        self.event_type = event_type
+        self.event_timestamp = now_timestamp()
+        for key, value in kwargs.items():
+            self.__setattr__(key, value)
+        # break soon if attributes are not valid
+        self.event_attributes()
+
+    def to_dict(self):
+        return {
+            "eventId": self.event_id,
+            "eventType": self.event_type,
+            "eventTimestamp": self.event_timestamp,
+            self._attributes_key(): self.event_attributes()
+        }
+
+    def _attributes_key(self):
+        key = "{0}EventAttributes".format(self.event_type)
+        return decapitalize(key)
+
+    def event_attributes(self):
+        if self.event_type == "WorkflowExecutionStarted":
```

Comment:
Sure, I'll do that! Honestly the logic is nearly always the same, it could probably be trimmed down to a dozen lines. At first I thought I would do a lot of complex formatting here, but in fact no.